mailbox/noise.go

```diff
@@ -761,12 +788,25 @@ func (b *Machine) RecvActTwo(actTwo [ActTwoSize]byte) error {
 		return err
 	}
 
-	payloadLen := binary.BigEndian.Uint16(payload[:2])
-	b.authData = make([]byte, payloadLen)
+	switch b.remoteVersion {
```

Comment:
Would be nice to add some unit tests for these version negotiations. I think adding a new functional option to create a client or server with a specific option should be pretty easy.

Reply:
do you mean that the functional option should specify the version?

and if so, but then we need to keep all the previous code too right? like for example, how will we test that the upgraded client works with the older server?

Or do you mean just create a unit test for each of the 'GenActX' functions and ensure that they return the correct data based on what is passed in?

Reply:
Agreed re tests here, only 4 combinations here so far that we can enumerate pretty easily. 

See comment above about awkwardness of sending the length of a future message vs just using the existing length prefix system to do so. If we just accept the packet size leak here, we can simplify by using a larger length int (uint32 instead of uint16) for the payload. And we do that _only_ for handshake v1, v0 uses the old 2 byte length prefix.